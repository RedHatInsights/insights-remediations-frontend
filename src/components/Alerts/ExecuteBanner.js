import React from 'react';
import propTypes from 'prop-types';

import { Alert, AlertActionLink } from '@patternfly/react-core';

const ExecuteBanner = ({ onCancel, ...props }) => {
    return (
        <Alert
            variant="default"
            isInline
            title="Playbook in progress"
            action={ <AlertActionLink onClick={ onCancel }>Cancel</AlertActionLink> }
            { ...props }
        />
    );
};

export default ExecuteBanner;

ExecuteBanner.propTypes = {
    onCancel: propTypes.func
};

/* eslint-disable no-console */
ExecuteBanner.defaultProps = {
    onCancel: () => console.log('Cancel Playbook')
};
/* eslint-enable no-console */
