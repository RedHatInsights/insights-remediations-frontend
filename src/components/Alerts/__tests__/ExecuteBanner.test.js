import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import ExecuteBanner from '../ExecuteBanner';

describe('ExecuteBanner component', () => {

    it('should render', () => {
        const wrapper = shallow(<ExecuteBanner/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    /* eslint-disable no-console */
    it('should click cancel with default prop', () => {
        console.log = jest.fn();
        const wrapper = mount(<ExecuteBanner/>);
        wrapper.find('AlertActionLink').simulate('click');
        expect(console.log.mock.calls[0][0]).toBe('Cancel Playbook');
    });
    /* eslint-enable no-console */

    it('should click cancel', () => {
        const mockCallBack = jest.fn();
        const wrapper = mount(<ExecuteBanner onCancel={ mockCallBack }/>);
        wrapper.find('AlertActionLink').simulate('click');
        expect(mockCallBack.mock.calls.length).toEqual(1);
    });
});
