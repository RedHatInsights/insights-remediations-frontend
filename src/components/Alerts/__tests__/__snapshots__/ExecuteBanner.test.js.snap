// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExecuteBanner component should render 1`] = `
<Component
  action={
    <AlertActionLink
      onClick={[Function]}
    >
      Cancel
    </AlertActionLink>
  }
  isInline={true}
  title="Playbook in progress"
  variant="default"
/>
`;
